inventory/inv_scripts/inventory_ui.gd

```diff
@@ -0,0 +1,31 @@
+extends Control
+
+@onready var inv: Inventory = preload("res://inventory/player_inventory.tres")
+@onready var slots: Array = $NinePatchRect/GridContainer.get_children()
```

Comment:
`: Array[InventorySlot]`

Reply:
and then you also need `$NinePatchRect/GridContainer.get_children() as Array[InventorySlot]`. these 2 things will ensure that it is statically typed

Reply:
infeasible